### DISCUSSION TITLE
Publish crashing with no error

### DISCUSSION THREAD

**Discussion by xrnss** [uNetworking/uWebSockets.js#368] [Q&A] [answered]
I know this is quite hard to crack the root on due to there being no error, however I will try my best to explain.

uWSApp starts -> Creates ChatChannel class with uWSApp as parameter, on listen event -> User sends chat message -> Class handles and publishes message -> App closes with no errors.

An interesting find is if I publish to a topic with no socket, the app does not crash.

Here are my code snippets:

```
 }).listen("0.0.0.0", 3000, sock => {
                this.listening = false;
                this.sock = sock;
                this.game.ChatChannel = new ChatChannel(this);
                console.log(`Server opened`);
                resolve(true);
            });
```

`this.uws.app.publish("broadcast", binaryArray, true);`

Any help appreciated, apologies if I'm being stupid here or something but there is absolutely 0 errors.

Edit: also to mention this.uws.app.publish is definitely defined as I logged it and it isn't undefined.

**Comment by hst-m** [comment]
can you simplify and remove enough code to post the full code of an example that still causes issue?

havent looked into this at all but something that could be related is your `sock` in listen callback https://unetworking.github.io/uWebSockets.js/generated/interfaces/us_listen_socket.html
> Native type representing a raw uSockets struct us_listen_socket. Careful with this one, it is entirely unchecked and native so invalid usage will blow up.

**Comment by xrnss** [comment]
Yeah I’ll get that for you tomorrow and look in to the testing without the sock being put there.

**Comment by ghost** [comment]
I need a lot more info to make any sense of this. It can be anything going wrong here - esp. if you make invalid usage of that listenSocket.

**Comment by ghost** [comment, marked as answer]
You haven't provided any info whatsoever. I've seen crashes in publish myself - they happen for instance if you pass wrong data types because of lacking type checking. That's probably your issue and I can't really bother with your issue report since you refuse to give any more info. Closing
